### PR TITLE
Add Checkbox for Publish to Project Catalogue

### PR DESCRIPTION
**Pull request by MightyAx**
Implementation of ebi-ait/dcp-ingest-central#284
Publish to the Human Cell Atlas Project Catalogue by EMBL-EBI:

- [x] Add the checkbox to both new and edit project forms
- [x] Default the checkbox to True for new projects (including those made from DOI)
- [x] Change layout for all form checkboxes to reduce duplication.

Checkbox Layout Before:
Screenshot 2021-05-25 at 16 02 16
Checkbox Layout After:
Screenshot 2021-05-25 at 16 16 11

Publish to the Human Cell Atlas Project Catalogue by EMBL-EBI Checkbox Examples:
Link in help text goes here: https://www.ebi.ac.uk/humancellatlas/project-catalogue/
Screenshot 2021-05-25 at 16 34 32
Screenshot 2021-05-25 at 16 34 50

### CODE REVIEW

File: client/src/app/metadata-schema-form/form-item/form-item.component.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div [ngClass]="rootClass">
   <div [ngClass]="{'label': true, 'read-only': data.disabled}">
-    <label class="vf-form__label">{{data.label}}</label>
-    <app-vf-asterisk *ngIf="!data.disabled" [isRequired]="data.isRequired"></app-vf-asterisk>
+    <label class="vf-form__label" *ngIf="data.disabled || !rootClass['vf-form__item--checkbox']">{{data.label}}</label>
```
Comment (amnonkhen) [inline client/src/app/metadata-schema-form/form-item/form-item.component.html:3]: I would make it DRYer, extract a function and use it here and in the next line.

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: client/src/app/metadata-schema-form/form-item/form-item.component.html (hunk 1 above)
Comment (MightyAx) [reply client/src/app/metadata-schema-form/form-item/form-item.component.html:3]: A function that would only be used once?
The next line is a different condition, should I make a function for that too?

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: client/src/app/project-registration/project-registration-form/project-registration-form.component.ts [hunk 2]
```diff
@@ -238,7 +241,8 @@ export class ProjectRegistrationFormComponent implements OnInit, OnDestroy {
             const publication = {};
 
             const projectFormData = {
-              content: {}
+              content: {},
```
Comment (jacobwindsor) [inline client/src/app/project-registration/project-registration-form/project-registration-form.component.ts:244]: Can you do `const projectFormData = { ...emptyProject }` instead?

**Review by jacobwindsor — CHANGES_REQUESTED** [review]

**Review by amnonkhen — APPROVED** [review]

**Review by aaclan-ebi — APPROVED** [review]

**Comment by yusra-haider** [comment]
merging this on behalf of @MightyAx since this has two approvals

**Pull request merged**